### ISSUE TITLE
mmr_lmproc.hist

### ISSUE THREAD

**Issue by sanazkaviani** [pjmark/NiftyPET#17]
Hello,
I have a problem with "mmr_lmproc.hist". Although I've installed the niftypet without any problem or error, my notebook crashes when I want to produce Sinogram. I am using "Tesla V100-SXM2-32GB" GPU.

**Comment by pjmark** [comment]
Can you please provide the error output?

Thanks,
Pawel

Pawel
________________________________
From: sanazkaviani ***@***.***>
Sent: Thursday, September 29, 2022 9:59:26 PM
To: NiftyPET/NiftyPET ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [NiftyPET/NiftyPET] mmr_lmproc.hist (Issue #17)


⚠ Caution: External sender


Hello,
I have a problem with "mmr_lmproc.hist". Although I've installed the niftypet without any problem or error, my notebook crashes when I want to produce Sinogram. I am using "Tesla V100-SXM2-32GB" GPU.

**Comment by sanazkaviani** [comment]
Thanks a lot for your answer,

I don't get errors; just the kernel dies as follows:
AsyncIOLoopKernelRestarter: restarting kernel (1/5), keep random ports
kernel 9834c3dc-9e86-4d39-9176-babf0908d46b restarted

I appreciate your help,
Sanaz

**Comment by pjmark** [comment]
Hi Sanaz,

Apologies for my delayed reply. I think the problem must be originating locally in your operating system - what system do you use?

Pawel

Pawel
________________________________

**Comment by sanazkaviani** [comment]
Hi Pawel,
Thanks a lot for your answer.

I'm using Compute Canada, which has a Linux environment. Niftypet has been installed successfully at Google Collab, but I would like to use it at compute Canada because I am developing deep-learning image reconstruction based on Niftypet, and I need a powerful system. 
The problem is that whenever I use '.so' libraries, it crashes. 

I would greatly appreciate your help in this regard.
Sanaz

**Comment by sanazkaviani** [comment]
Hi Pawel,

Hope you are doing well,

Regarding my last question, I would greatly appreciate it if you could answer me so that we can resolve this issue.

Bests,
Sanaz

**Comment by pjmark** [comment]
Unfortunately, I don't have experience with those systems.

But going from the beginning: are you able to successfully install NiftyPET, including compilation? Also, are you able to run a terminal and import it?

Cheers
Pawel

Pawel
________________________________

**Comment by sanazkaviani** [comment]
Yes, the installation was completely successful, without any errors. Also, I could import the Niftypet, Nipet, and everything then I could get the mmrparams, but when I want to get sinograms with the mmrhist it crashes.

Sanaz

**Comment by pjmark** [comment]
What happens of you go to terminal and run nipet.gpuinfo()?

Pawel
________________________________

**Comment by sanazkaviani** [comment]
Sorry for my late reply, allocating the kernel takes time.
I receive my GPU information as follows:
[('Tesla V100-SXM2-32GB', 32768, 7, 0)]

**Comment by sanazkaviani** [comment]
I'd appreciate it if you could provide me with some information about ".so" libraries (how can I open and edit them) because I think my problem is them (E.g : mmr_lmproc.cpython-37m-x86_64-linux-gnu.so , mmr_auxe.so, and ...)

**Comment by sanazkaviani** [comment]
Hello again Pawel,

my Linux is 'CentOS 7', do you think that maybe the problem is because of my Linux?
